### PR TITLE
Tempest Keep Update

### PR DESCRIPTION
**Pull request by Qingle**
### - Movement has been improved, has the chance to do one of 3 things:
1. Stay at the current platform and summon an ember.
2. Move to the next platform and summon an ember.
3. Move to the middle of the room and Flame Quill.

### - Updated the threat system to do the following:
1. Check first if there are any melee in range.
2. If no melee are in range, target the person at range with the highest threat.
3. Use Flame Buffet if they are out of range for auto attack.

## Void Reaver:
- Fixed a bug where he **wouldn't** use Arcane Orb if everyone is within 18 yards.

## High Astromancer Solarian:
- Implemented the AoE arcane damage for Wrath of the Astromancer.
- Made it so that she does Mark of Solarian on every Blinding Light.
- Fixed Timers.

Note: Squashed to one.

### CODE REVIEW

**Comment by xfurry** [comment]
Poke @killerwife to review.
I can suggest some small improvements, but I don't have time to go into details at the moment.

**Comment by killerwife** [comment]
Actually I was the one who gave him the incentive to look at it. This goes hand in hand with the complete SQL revamp of TK. So yeah, I will provide feedback once I am done with my school/work on the attunement.

**Comment by Qingle** [comment]
I think I have an idea of how it can be improved, especially with the Solarian AoE.

The dummy has a range, so if I can update Spell.cpp to use the range to get a targetList, would that work?
As well as that I could use the SpellEffects.cpp#L323 to do the damage split?

**Comment by robinsch** [comment]
No it's not EffectDMG.

**Comment by Qingle** [comment]
Okay thank you for the insight, any specific improvements you can think of robinsch?
I'm new to this stuff still :)

File: src/scriptdev2/scripts/outland/tempest_keep/the_eye/boss_alar.cpp [hunk 1]
```diff
@@ -156,17 +157,64 @@ struct boss_alarAI : public ScriptedAI
                 pSummoned->AI()->AttackStart(pTarget);
         }
     }
-
+    
     void SummonedCreatureJustDied(Creature* pSummoned) override
     {
-        // drain 3% of boss health when the ember dies
+        // drain 2% of boss health when the ember dies
```
Comment (Patrick-Edouard) [inline src/scriptdev2/scripts/outland/tempest_keep/the_eye/boss_alar.cpp]: You changed the comment, but not the value in the code bellow.

According to wow wiki it's indeed 2% and not 3.

**Comment by killerwife** [comment]
@Qingle I dont see anything inherently wrong about any of your changes. Only disputable thing is that Wrath of the Astromancer Dot implementation, but not because of your code, but because we know so little about it.

**Comment by Qingle** [comment]
@killerwife the problem i see about the DoT is that it doesn't actually show up on the combat log, which means it may be implemented in a different way. I'll do some digging and see if there is another way to go about it

**Pull request closed without merging**